tasks/queue-granules/index.js

```diff
@@ -62,6 +120,69 @@ function updateGranuleBatchCreatedAt(granuleBatch, createdAt) {
   }));
 }
 
+/**
+ * Constructs the partial params for enqueueGranuleIngestMessage. This is an optimization.
+ * This part of the message requires loading files out of S3. We don't want to do that everytime
+ * we create an SQS message considering that these parts at the same for every message, so this
+ * is done before looping through the chunks of granules as an optimization.
+ */
+async function buildPartialMessage({
+  granuleIngestWorkflow,
+  stack,
+  systemBucket,
+}) {
+  const messageTemplate = await getJsonS3Object(systemBucket, templateKey(stack));
+  const { arn } = await getJsonS3Object(
+    systemBucket,
+    getWorkflowFileKey(stack, granuleIngestWorkflow)
+  );
+  return {
+    messageTemplate,
+    workflow: {
+      name: granuleIngestWorkflow,
+      arn,
+    },
+  };
+}
+
+/**
+ * This mimics the function of enqueueGranuleIngestMessage but takes a partially
+ * constructed message as an additional param.
+ *
+ * @see buildPartialMessage()
+ */
+async function enqueueGranuleIngestMessage({
+  collection,
+  granules,
+  parentExecutionArn,
+  pdr,
+  provider,
+  partialMessage,
+  queueUrl,
+  executionNamePrefix,
+  additionalCustomMeta = {},
+}) {
+  const message = buildQueueMessageFromTemplate({
+    ...partialMessage,
+    parentExecutionArn,
+    payload: { granules },
+    customMeta: {
+      ...additionalCustomMeta,
+      ...(pdr ? { pdr } : {}),
+      collection,
+      provider,
+    },
+    executionNamePrefix,
+  });
+
+  await sendSQSMessage(queueUrl, message);
+
+  return buildExecutionArn(
+    message.cumulus_meta.state_machine,
+    message.cumulus_meta.execution_name
+  );
+}
```

Comment:
Is there a reason we shouldn't just update the function in the ingest package?

Reply:
yea def could do that, but I would do that refactor separately in all tasks rather than growing this PR to do that.

Reply:
We should update `enqueueGranuleIngestMessage` in the ingest package to take `granuleIngestWorkflowArn` and `cumulusMessageTemplate`, and that function is only used in `queue-granules` package for now.